File: app/src/main/java/com/gmail/vanyadubik/freeride/utils/ScreenUtils.java
package com.gmail.vanyadubik.freeride.utils;

import android.app.Activity;
import android.content.Context;
import android.util.DisplayMetrics;
import android.view.Display;
import android.view.View;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.inputmethod.InputMethodManager;
import com.gmail.vanyadubik.freeride.R;

public class ScreenUtils {

    Context ctx;
    DisplayMetrics metrics;

    public ScreenUtils(Context ctx) {
        this.ctx = ctx;
        WindowManager wm = (WindowManager) ctx
                .getSystemService(Context.WINDOW_SERVICE);

        Display display = wm.getDefaultDisplay();
        metrics = new DisplayMetrics();
        display.getMetrics(metrics);

    }

    public int getHeight() {
        return metrics.heightPixels;
    }

    public int getWidth() {
        return metrics.widthPixels;
    }

    public int getRealHeight() {
        return metrics.heightPixels / metrics.densityDpi;
    }

    public int getRealWidth() {
        return metrics.widthPixels / metrics.densityDpi;
    }

    public int getDensity() {
        return metrics.densityDpi;
    }

    public int getScale(int picWidth) {
        Display display
                = ((WindowManager) ctx.getSystemService(Context.WINDOW_SERVICE))
                .getDefaultDisplay();
        int width = display.getWidth();
        Double val = new Double(width) / new Double(picWidth);
        val = val * 100d;
        return val.intValue();
    }

    public static void closeKeyboard(Context context) {
        View view = ((Activity)context).getCurrentFocus();
        if (view != null) {
            InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
            if (imm != null) {
                imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
            }
        }
    }

    public static void slideViewUp(Context context, View view){

       Animation slide_up = AnimationUtils.loadAnimation(context.getApplicationContext(),
                R.anim.slide_up);

        view.startAnimation(slide_up);
        slide_up.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                view.setVisibility(View.GONE);
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });

    }
    public static void slideViewDown(Context context, View view){

        view.setVisibility(View.VISIBLE);

        Animation slide_down = AnimationUtils.loadAnimation(context.getApplicationContext(),
                R.anim.slide_down);

        view.startAnimation(slide_down);
    }


}

File: app/src/main/java/com/gmail/vanyadubik/freeride/utils/ScreenUtils.kt
package com.gmail.vanyadubik.freeride.utils

import android.app.Activity
import android.content.Context
import android.util.DisplayMetrics
import android.view.Display
import android.view.View
import android.view.WindowManager
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.view.inputmethod.InputMethodManager
import com.gmail.vanyadubik.freeride.R

class ScreenUtils(internal var ctx: Context) {
    internal var metrics: DisplayMetrics

    val height: Int
        get() = metrics.heightPixels

    val width: Int
        get() = metrics.widthPixels

    val realHeight: Int
        get() = metrics.heightPixels / metrics.densityDpi

    val realWidth: Int
        get() = metrics.widthPixels / metrics.densityDpi

    val density: Int
        get() = metrics.densityDpi

    init {
        val wm = ctx
            .getSystemService(Context.WINDOW_SERVICE) as WindowManager

        val display = wm.defaultDisplay
        metrics = DisplayMetrics()
        display.getMetrics(metrics)

    }

    fun getScale(picWidth: Int): Int {
        val display = (ctx.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
            .defaultDisplay
        val width = display.width
        var `val`: Int = width / picWidth
        `val` = (`val`!! * 100.0).toInt()
        return `val`.toInt()
    }

    companion object {

        fun closeKeyboard(context: Context) {
            val view = (context as Activity).currentFocus
            if (view != null) {
                val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                imm?.hideSoftInputFromWindow(view.windowToken, 0)
            }
        }

        fun slideViewUp(context: Context, view: View) {

            val slide_up = AnimationUtils.loadAnimation(
                context.applicationContext,
                R.anim.slide_up
            )

            view.startAnimation(slide_up)
            slide_up.setAnimationListener(object : Animation.AnimationListener {
                override fun onAnimationStart(animation: Animation) {

                }

                override fun onAnimationEnd(animation: Animation) {
                    view.visibility = View.GONE
                }

                override fun onAnimationRepeat(animation: Animation) {

                }
            })

        }

        fun slideViewDown(context: Context, view: View) {

            view.visibility = View.VISIBLE

            val slide_down = AnimationUtils.loadAnimation(
                context.applicationContext,
                R.anim.slide_down
            )

            view.startAnimation(slide_down)
        }
    }


}